src/Microsoft.Diagnostics.ExtensionCommands/DumpAsyncCommand.cs

```diff
@@ -0,0 +1,1166 @@
+                            WriteLine($" {Truncate(field.Type.Name, 32),-32} {Truncate(GetDisplay(obj, field).ToString(), 16),16} {field.Name}");
+                        }
+                    }
+                }
+            }
+
+            // <summary>Gets a printable description for the specified object.</summary>
+            string Describe(ClrObject obj)
+            {
+                // Default the description to the type name.
+                string description = obj.Type.Name;
+
+                if (IsStateMachineBox(obj.Type))
+                {
+                    // Remove the boilerplate box type from the name.
+                    int pos = description.IndexOf("StateMachineBox<", StringComparison.Ordinal);
+                    if (pos >= 0)
+                    {
+                        ReadOnlySpan<char> slice = description.AsSpan(pos + "StateMachineBox<".Length);
+                        slice = slice.Slice(0, slice.Length - 1); // remove trailing >
+                        description = slice.ToString();
+                    }
+                }
+                else if (TryGetValidObjectField(obj, "m_action", out ClrObject taskDelegate))
+                {
+                    // If we can figure out what the task's delegate points to, append the method signature.
+                    if (TryGetMethodFromDelegate(runtime, taskDelegate, out ClrMethod? method))
+                    {
+                        description = $"{description} {{{method!.Signature}}}";
+                    }
+                }
+
+                return description;
+            }
+
+            // <summary>Determines whether the specified object is of interest to the user based on their criteria provided as command arguments.</summary>
+            bool IncludeInOutput(ClrObject obj)
+            {
+                if (ObjectAddress is ulong addr && obj.Address != addr)
+                {
+                    return false;
+                }
+
+                if (MethodTableAddress is ulong mt && obj.Type.MethodTable != mt)
+                {
+                    return false;
+                }
+
+                if (NameSubstring is not null && !obj.Type.Name.Contains(NameSubstring))
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            // <summary>Finds all of the relevant async-related objects on the heap.</summary>
+            Dictionary<ClrObject, AsyncObject> CollectObjects()
+            {
+                var found = new Dictionary<ClrObject, AsyncObject>();
+
+                // Enumerate the heap, looking for all relevant objects.
+                foreach (ClrObject obj in heap.EnumerateObjects())
+                {
+                    Console.CancellationToken.ThrowIfCancellationRequested();
+
+                    if (!obj.IsValid || obj.Type is null)
+                    {
+                        WriteLineError($"(Skipping invalid object {obj})");
```

Comment:
```suggestion
                        Trace.WriteError($"(Skipping invalid object {obj})");
```

Reply:
Where is Trace routed?

Reply:
There's a command that sets a listener if requested for things like this. If it's potentially verbose and such, or informational it usually goes to those logs. The other errors i saw were a one time print thing that could tell the user something went wrong. Seeing a bunch of heap corruption messages didn't seem as useful.

Reply:
Ok, cool. For reference, what command?

Reply:
!logging enable/disable